app/View/Components/Forms/Datalist.php

```
@@ -47,11 +51,27 @@ public function __construct(
+            $this->modelBinding = 'null';
         }
+
+        // Generate unique HTML ID by adding random suffix
+        // This prevents duplicate IDs when multiple forms are on the same page
+        if ($this->modelBinding && $this->modelBinding !== 'null') {
+            // Use original ID with random suffix for uniqueness
+            $uniqueSuffix = substr(md5(uniqid((string) mt_rand(), true)), 0, 8);
+            $this->htmlId = $this->modelBinding.'-'.$uniqueSuffix;
+        } else {
+            $this->htmlId = (string) $this->id;
+        }
```

PHPMD (2.15.0)

The method render uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)